ccm_web/client/src/pages/AddUMUsers.tsx

```diff
@@ -0,0 +1,450 @@
+      paddingTop: '20px'
+    },
+    uploadContainer: {
+      position: 'relative',
+      zIndex: 0,
+      textAlign: 'center'
+    },
+    dialog: {
+      textAlign: 'center',
+      marginBottom: 15,
+      paddingLeft: 10,
+      paddingRight: 10
+    },
+    table: {
+      paddingLeft: 10,
+      paddingRight: 10
+    },
+    dialogIcon: {
+      color: '#3F648E'
+    },
+    parseErrorDialog: {
+      textAlign: 'center',
+      marginBottom: 15,
+      paddingLeft: 10,
+      paddingRight: 10
+    },
+    parseErrorTable: {
+      paddingLeft: 10,
+      paddingRight: 10
+    },
+    parseErrorIcon: {
+      color: 'red'
+    },
+    sectionLoadErrorIcon: {
+      color: '#3F648E'
+    },
+    sectionLoadError: {
+      textAlign: 'center'
+    }
+  })
+)
+
+interface AddUMUsersProps extends CCMComponentProps {}
+
+function AddUMUsers (props: AddUMUsersProps): JSX.Element {
+  enum States {
+    SelectSection = 0,
+    UploadCSV = 1,
+    ReviewCSV = 2
+  }
+
+  const classes = useStyles()
+  const [activeStep, setActiveStep] = useState(States.SelectSection)
+
+  const [sections, setSections] = useState<CanvasCourseSection[]>([])
+  const [file, setFile] = useState<File|undefined>(undefined)
+  const [enrollments, setEnrollments] = useState<IAddUMUserEnrollment[]|undefined>(undefined)
+  const [errors, setErrors] = useState<ValidationError[]|undefined>(undefined)
+  const [isAddingEnrollments, setIsAddingEnrollments] = useState<boolean>(false)
+
+  const updateSections = (sections: CanvasCourseSection[]): void => {
+    setSections(sections.sort((a, b) => { return a.name.localeCompare(b.name) }))
+  }
+
+  const [doLoadCanvasSectionData, isExistingSectionsLoading, getCanvasSectionDataError] = usePromise(
+    async () => await getCourseSections(props.globals.course.id),
+    (sections: CanvasCourseSection[]) => {
+      updateSections(sections)
+    }
+  )
+
+  useEffect(() => {
+    void doLoadCanvasSectionData()
+  }, [])
+
+  useEffect(() => {
+    if (file === undefined) {
+      // ?
+    } else {
+      parseFile(file)
+    }
+  }, [file])
+
+  const getSteps = (): string[] => {
+    return ['Select', 'Upload', 'Review', 'Confirmation']
+  }
+  const steps = getSteps()
+
+  const [selectedSections, setSelectedSections] = useState<CanvasCourseSection[]>([])
+
+  const sectionCreated = (newSection: CanvasCourseSection): void => {
+    updateSections(sections.concat(newSection))
+    setSelectedSections([newSection])
+  }
+
+  const isValidRole = (role: string): boolean => {
+    return canvasRoles.map(canvasRole => { return canvasRole.clientName.localeCompare(role, 'en', { sensitivity: 'base' }) === 0 }).filter(value => { return value }).length > 0
+  }
+
+  const isValidLoginID = (loginID: string): boolean => {
+    // return uniqname.match(UNIQNAME_REGEX) !== null
+    // Don't apply any validation here, just pass anything through
+    // Flag this function for deletion in 3.. 2.. 1...
+    return true
+  }
```

Comment:
Should this code be removed? since it just returning true

Reply:
I'm sure the reviewers will vote yes.  I left it there b/c it seemed reasonable that we might want to do some validation of the text.. I had been applying the aforementioned uniqname regex.  Maybe there is nothing to do  here ever.

Reply:
Is there any value that is absolutely 100% always illegal every single time in the client?